### PR TITLE
Move simple-game-server to Distroless base

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

/kind cleanup

> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

This includes:
* Replacing Alpine with Distroless as the base image
* Tweaked the creation of the mulit-arch image manifest to use a more recent command (something changed in Docker 24.0.4 that broke the current setup, also this is less steps).
* Updated all the reference to simple-game-server:0.16 to simple-game-server:0.17.

**Which issue(s) this PR fixes**:

Work on #909

**Special notes for your reviewer**:

First run of this will fail, as I haven't pushed up the image. Wanted to get a set of eyes on it first before pushing up the new version.

Local testing has this passing the e2e tests I manually ran (not the full suite), so I'm reasonably confident 😄 .

### CODE REVIEW

**Review by gongmax — APPROVED** [review]

**Review by roberthbailey — APPROVED** [review]

**Comment by markmandel** [comment]
Ooh that's a fun flake. Looks like a feature flag flipped mid way.

```
{"message":"k8s.io/client-go/informers/factory.go:150: Failed to watch *v1.Secret: unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"\", Version:\"v1\", Resource:\"secrets\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}}","severity":"error","time":"2023-07-20T00:24:56.466041815Z"}
--- FAIL: TestAllocatorAllocatePriority (1.18s)
    allocator_test.go:169: 
        	Error Trace:	/go/src/agones.dev/agones/pkg/gameserverallocations/allocator_test.go:169
        	            				/go/src/agones.dev/agones/pkg/gameserverallocations/allocator_test.go:176
        	Error:      	"[spec.selectors[0].counters: Forbidden: Feature CountsAndLists must be enabled spec.selectors[0].lists: Forbidden: Feature CountsAndLists must be enabled]" should have 0 item(s), but has 2
        	Test:       	TestAllocatorAllocatePriority
```

**Comment by igooch** [comment]
> TestAllocatorAllocatePriority

That's odd -- that test isn't even for Counts and Lists Priorities.

**Comment by markmandel** [comment]
> > TestAllocatorAllocatePriority
> 
> That's odd -- that test isn't even for Counts and Lists Priorities.

See PR #3280 - I reckon the feature flag flipped RIGHT between the `ApplyDefaults` and Validate` calls.

File: examples/simple-game-server/Makefile [hunk 1]
```diff
@@ -82,9 +82,13 @@ endif
 ifeq ($(WITH_ARM64), 1)
 push: push-linux-image-arm64
 endif
-	-DOCKER_CLI_EXPERIMENTAL=enabled docker manifest rm $(server_tag)
-	DOCKER_CLI_EXPERIMENTAL=enabled docker manifest create $(server_tag) $(push_server_manifest)
-	DOCKER_CLI_EXPERIMENTAL=enabled docker manifest push $(server_tag) --purge
+	-docker manifest rm $(server_tag)
+	docker manifest create $(server_tag) $(push_server_manifest)
+	docker manifest push $(server_tag) --purge
+	# TODO: Update to replace `docker manifest` commands if/when the above commands fail (~Docker 24.x)
```
Comment (markmandel) [inline examples/simple-game-server/Makefile:88]: The above commands are still marked as "experimental" but work in Docker 20.x (which is what we have locally and on Cloud Build).  `docker buildx imagetools create` has a bug in it with GCR/Artifact registry though.

I tested on Docker 24.x on a different machine where `docker manifest` now breaks (it is experimental) but `docker buildx imagetools create` works. So leaving it for when the versions eventually get updated and we'll need a reference on how to fix things.

**Review by roberthbailey — APPROVED** [review]

**Pull request merged**